src/type-checker.cc

```diff
@@ -598,24 +604,24 @@ Result TypeChecker::OnLoop(const TypeVector& param_types,
   return result;
 }
 
-Result TypeChecker::OnMemoryCopy() {
-  return CheckOpcode3(Opcode::MemoryCopy);
+Result TypeChecker::OnMemoryCopy(const Limits& limits) {
+  return CheckOpcode3(Opcode::MemoryCopy, &limits, &limits, &limits);
 }
 
 Result TypeChecker::OnDataDrop(uint32_t segment) {
   return Result::Ok;
 }
 
-Result TypeChecker::OnMemoryFill() {
-  return CheckOpcode3(Opcode::MemoryFill);
+Result TypeChecker::OnMemoryFill(const Limits& limits) {
+  return CheckOpcode3(Opcode::MemoryFill, &limits, nullptr, &limits);
 }
 
 Result TypeChecker::OnMemoryGrow() {
   return CheckOpcode1(Opcode::MemoryGrow);
 }
 
-Result TypeChecker::OnMemoryInit(uint32_t segment) {
-  return CheckOpcode3(Opcode::MemoryInit);
+Result TypeChecker::OnMemoryInit(uint32_t segment, const Limits& limits) {
+  return CheckOpcode3(Opcode::MemoryInit, &limits, &limits, &limits);
```

Comment:
I think we want the size and source offset to be `i32` still, since the data segments are still limited to 32-bits.

Reply:
This particular piece of code does not have any 32->64 bit changes, what code are your referring to?

Reply:
At least in LLVM all 3 operands are now 64-bit, so that would need fixing too then: https://reviews.llvm.org/D82821
I guess I wasn't aware these can only come from data segments, and that we have no plans to make those 64-bit ever..

Reply:
Are the other bulk memory ops correct?

Reply:
I think so, I just wrote up the new ones here: https://github.com/WebAssembly/memory64/blob/master/proposals/memory64/Overview.md#validation